docs/flow/reference/components/discovery.kubernetes.md

```diff
@@ -0,0 +1,164 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/discovery.kubernetes
+title: discovery.kubernetes
+---
+
+# discovery.kubernetes
+
+`discovery.kubernetes` allows you to find scrape targets from Kubernetes resources. It watches cluster state, and ensures targets are continually synced with what is currently running in your cluster.
+
+Internally, this component uses Kubernetes Service Discovery from Prometheus. [The documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config) on their site offers further details on the metadata labels applied to targets and other usage considerations.
+
+If you supply no connection information, this component defaults to using an in-cluster config. Otherwise, you can specify a kubeconfig file, or manual apiserver configuration.
```

Comment:
```suggestion
If you supply no connection information, this component defaults to an in-cluster config. Otherwise, you can specify a kubeconfig file or a manual apiserver configuration.
```